src/Diagnostics.DataProviders/KustoSDKClient.cs

```diff
@@ -111,15 +112,40 @@ public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, str
                                     shadowClientRequestProperties.SetOption(ClientRequestProperties.OptionQueryConsistency, ClientRequestProperties.OptionQueryConsistency_Strong);
                                 }
                                 var shadowKustoClient = Client(shadowCluster, database);
-                                shadowKustoClient.ExecuteQueryAsync(database, query, shadowClientRequestProperties);
+                                kustoTask = shadowKustoClient.ExecuteQueryAsync(database, query, shadowClientRequestProperties)

```

Comment:
@yifguoMSFT , will this create a log in the DiagnosticRole table when the query is fired against the new leader cluster?

Reply:
It should, as I changed the cluster variable in line 125 if the task succeeds and will be logged in line 171. But I am not very sure what will happen after `LogKustoQuery` is called at line 171, just assumed it will create a log